src/thingset_bin.c

```diff
@@ -141,6 +161,44 @@ static int bin_serialize_path(struct thingset_context *ts,
                : -THINGSET_ERR_RESPONSE_TOO_LARGE;
 }
 
+static int bin_serialize_metadata(struct thingset_context *ts,
+                                  const struct thingset_data_object *object)
+{
+    int err = bin_serialize_map_start(ts);
+    if (err) {
+        return err;
+    }
+
+    const char *name = "name";
+    if (!zcbor_tstr_encode_ptr(ts->encoder, name, 4)) {
```

Comment:
Suggestion to avoid magic numbers. This will be evaluated at compile time and does not introduce any overhead:
```suggestion
    if (!zcbor_tstr_encode_ptr(ts->encoder, name, sizeof(name) - 1)) {
```